File: back-end/src/main/java/com/ocire/lms/service/ForumService.java
package com.ocire.lms.service;

import java.util.List;

import com.ocire.lms.dto.InsertResDto;
import com.ocire.lms.dto.forum.CommentInsertReqDto;
import com.ocire.lms.dto.forum.CommentResDto;
import com.ocire.lms.model.Forum;

public interface ForumService {
	Forum getForumByLearningId(Long learningId);

	Forum insertForum(Forum forum);
	
	List<CommentResDto> getCommentByForumId(Long forumId);

	InsertResDto insertComment(CommentInsertReqDto data);
}

File: front-end/src/app/services/forum.service.ts
import { Injectable } from '@angular/core';
import { BaseService } from './base.service';
import { Observable } from 'rxjs';
import { CommentResDto } from '@dto/comment/comment.res.dto';
import { BASE_URL } from '../constant/api.constant';
import { CommentInsertReqDto } from '@dto/comment/comment.insert.req.dto';
import { InsertResDto } from '@dto/InsertResDto';

@Injectable({
  providedIn: 'root',
})
export class ForumService {
  constructor(private base: BaseService) {}

  getCommentByForumId(id: number): Observable<CommentResDto[]> {
    return this.base.get(`${BASE_URL}/forum/comment/?id=${id}`);
  }

  insert(data: CommentInsertReqDto): Observable<InsertResDto> {
    return this.base.post(`${BASE_URL}/forum`, data);
  }
}
